### DISCUSSION TITLE
How to configure when running in the same Kubernetes cluster as the auth provider

### DISCUSSION THREAD

**Discussion by nilskaspersson** [nextauthjs/next-auth#6405] [Help]
Hi all,

My environment must run in a Kubernetes cluster, where one pod hosts Next.js + NextAuth, and another pod hosts Keycloak. When Running the Next.js app locally (outside of the cluster), the auth flow works as expected, but it does not when the two pods are in the same cluster.

I set the `NEXTAUTH_URL_INTERNAL` to the ``http://${NEXTJS_POD_IP_ADDRESS}``, however it still errors when attempting to construct the authentication URL. This is the error + stack trace that I get:

```bash
[next-auth][error][SIGNIN_OAUTH_ERROR]
https://next-auth.js.org/errors#signin_oauth_error outgoing request timed out after 3500ms {
  error: {
    message: 'outgoing request timed out after 3500ms',
    stack: 'RPError: outgoing request timed out after 3500ms\n' +
      '    at /home/node/app/node_modules/openid-client/lib/helpers/request.js:137:13\n' +
      '    at async Issuer.discover (/home/node/app/node_modules/openid-client/lib/issuer.js:144:24)\n' +
      '    at async openidClient (/home/node/app/node_modules/next-auth/core/lib/oauth/client.js:16:14)\n' +
      '    at async getAuthorizationUrl (/home/node/app/node_modules/next-auth/core/lib/oauth/authorization-url.js:67:18)\n' +
      '    at async Object.signin (/home/node/app/node_modules/next-auth/core/routes/signin.js:38:24)\n' +
      '    at async AuthHandler (/home/node/app/node_modules/next-auth/core/index.js:253:26)\n' +
      '    at async NextAuthHandler (/home/node/app/node_modules/next-auth/next/index.js:23:19)\n' +
      '    at async /home/node/app/node_modules/next-auth/next/index.js:59:32\n' +
      '    at async Object.apiResolver (/home/node/app/node_modules/next/dist/server/api-utils/node.js:363:9)\n' +
      '    at async NextNodeServer.runApi (/home/node/app/node_modules/next/dist/server/next-server.js:487:9)',
    name: 'RPError'
  },
  providerId: 'keycloak',
  message: 'outgoing request timed out after 3500ms'
```

Where the function at `/openid-client/lib/helpers/request.js:137:13` attempts to request ``${options.issuer}/.well-known/openid-configuration``, but times out.

Now, the `options.issuer` is of course configured to be the public issuer domain, which is not accessible from within the pod. 

How _should_ this be configured? It seems to me like I would need to query ``http://${KEYCLOAK_POD_IP_ADDRESS}/.well-known/openid-configuration`` instead, but I cannot configure it as such.

Hosting the auth provider and the app on the same cluster doesn't strike me as a very unique setup, so I'm surprised that there is no configuration option here. Am I misunderstanding something? To add to my confusion, this configuration setup has worked in the past, and is functional at seemingly random time intervals.

**Comment by SeoYoung-C** [comment]
I'm experiencing a similar issue. I'm using Okta as the identity provider, and I'm not sure how to resolve it. Can you provide any guidance or suggestions on troubleshooting this?
